sdk/monitor/monitor-ingestion/src/logsIngestionClient.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential } from "@azure/core-auth";
+import { CommonClientOptions } from "@azure/core-client";
+import { SDK_VERSION } from "./constants";
+import { GeneratedDataCollectionClient } from "./generated";
+import { FailedLogsIngestionError, UploadOptions, UploadResult } from "./models";
+import { GZippingPolicy } from "./gZippingPolicy";
+import asyncPool from "tiny-async-pool";
```

Comment:
can we avoid this dependency? Storage-blob has its own "BufferScheduler" class to handle this kind of thing, but I believe you should be able to pare down a minimal version to handle parallel upload of buffers

Reply:
Removed the dependency.